tests/reqres.rs

```diff
@@ -0,0 +1,98 @@
+use asynchronous_codec::{Event, LinesCodec, RecvSend};
+use futures::channel::oneshot;
+use futures::io::Cursor;
+use futures_util::stream::SelectAll;
+use futures_util::{FutureExt, StreamExt};
+use std::iter::FromIterator;
+use std::time::Duration;
+
+#[test]
+fn smoke() {
+    let mut buffer = Vec::new();
+    buffer.extend_from_slice(b"hello\n");
+
+    let mut stream = RecvSend::new(Cursor::new(&mut buffer), LinesCodec).close_after_send();
```

Comment:
The buffer is meant to simulate a network stream.

Reply:
Yes, I just found example counter-intuitive because TCP stream (as an example) is duplex, but reads and writes are accumulated on different sides rather than being concatenated in the same vector. Composing duplex stream from separate read and write buffers would be more intuitive, but it would also be much more code.

I guess what I anticipated is some comment about why we read and write to the same slice of memory.

Reply:
I see what you are saying. Unfortunately, I am yet to find a good test utility that plugs neatly into `AsyncRead` + `AsyncWrite`. Might need to write one ...